### PR TITLE
fix: azure Lacework provider & profile flag

### PR DESCRIPTION
**Pull request by jon-stewart**
## Summary

Azure TF requires the Lacework provider and the Lacework API profile defined, if a default profile hasn't been configured.

```
> terraform plan
╷
│ Error: Invalid provider configuration
│
│ Provider "registry.terraform.io/lacework/lacework" requires explicit configuration. Add a provider block to the root module and configure the provider's required arguments as described in
│ the provider documentation.
│
╵
╷
│ Error: Unable to create Lacework API client
│
│   with provider["registry.terraform.io/lacework/lacework"],
│   on <empty> line 0:
│   (source code not available)
│
│ profile 'default' not found.
│
│ Try using the Lacework CLI command 'lacework configure --profile default'
```

This error doesn't occur if `.lacework.toml` contains a default block.  It is a problem for the onboarding service as the bundle (bash script) doesn't configure a default profile, it configures a profile called 'onboarding'.

## How did you test this change?

- unit
- integration
- manual

## Issue

[RAIN-38163](https://lacework.atlassian.net/browse/RAIN-38163)

### CODE REVIEW

File: lwgenerate/azure/azure.go [hunk 1]
```diff
@@ -242,6 +256,11 @@ func (args *GenerateAzureTfConfigurationArgs) Generate() (string, error) {
 
 func createRequiredProviders() (*hclwrite.Block, error) {
 	return lwgenerate.CreateRequiredProviders(
+		lwgenerate.NewRequiredProvider(
+			"lacework",
+			lwgenerate.HclRequiredProviderWithSource(lwgenerate.LaceworkProviderSource),
+			lwgenerate.HclRequiredProviderWithVersion(lwgenerate.LaceworkProviderVersion),
```
Comment (rmoles) [inline lwgenerate/azure/azure.go:262]: I wonder if we can bump this Lacework provider version. Latest is 0.26 https://github.com/lacework/terraform-provider-lacework/tags

**Review by rmoles — APPROVED** [review]
Two comments. Otherwise, this looks good.

File: lwgenerate/azure/azure.go (hunk 1 above)
Comment (rmoles) [reply lwgenerate/azure/azure.go:262]: Do we also need to implement this for AWS/GCP?

Comment (jon-stewart) [reply lwgenerate/azure/azure.go:262]: AWS and GCP already had the lacework provider section in place, this issue only affecting Azure when running `terraform plan/apply`

**Pull request merged**